src/stream/ngx_stream_mruby_connection.c

```diff
@@ -52,36 +51,42 @@ static mrb_value ngx_stream_mrb_connection_init(mrb_state *mrb, mrb_value self)
 
   ctx = (ngx_stream_mruby_upstream_context *)mrb_malloc(mrb, sizeof(ngx_stream_mruby_upstream_context));
 
-  ctx->upstream = upstream;
   ctx->target = NULL;
   ctx->peers = NULL;
   ctx->us = NULL;
 
+  ctx->ictx = (ngx_stream_mruby_internal_ctx_t *)mrb_malloc(mrb, sizeof(ngx_stream_mruby_internal_ctx_t));
+  ctx->ictx->s = (ngx_stream_session_t *)mrb_malloc(mrb, sizeof(ngx_stream_session_t));
+
+  *ctx->ictx->s = *((ngx_stream_mruby_internal_ctx_t *)mrb->ud)->s;
```

Comment:
Why did you copy the dereference data to ictx->s? I think we use only the pointer.

Reply:
sorry.i fix it.